UnityProject/Assets/Editor/LevelBuilder/GUI/LevelBuilderWindow.cs

```diff
@@ -102,53 +94,91 @@ public void OnGUI()
 		if (SceneManager.GetActiveScene().name.Equals("Main"))
 			return;
 
-		if (GUILayout.Button (new GUIContent ("Create a new level."))) {
+		if (GUILayout.Button (new GUIContent ("Create a new level.")))
+        {
 
-			// We have a bug where Unity doesn't recognise our dynamically 
-			// added objects through the level builder so we need to force
-			// the change to be saved.
-			EditorSceneManager.MarkAllScenesDirty();
-			EditorSceneManager.SaveOpenScenes ();
+            // We have a bug where Unity doesn't recognise our dynamically 
+            // added objects through the level builder so we need to force
+            // the change to be saved.
+            EditorSceneManager.MarkAllScenesDirty();
+            EditorSceneManager.SaveOpenScenes();
 
-			Scene newScene = SceneHandler.createScene();
+            Scene newScene = SceneHandler.createScene();
 
-			// Unload the scenes so we can create a new one.
-			SceneHandler.unloadScenes();
+            // Unload the scenes so we can create a new one.
+            SceneHandler.unloadScenes();
 
-			// Calling countScenes() inside SceneHandler to find the new 
-			// filename to save our scene with. 
-			int currentNoOfLevels = SceneHandler.countScenes ();
-			currentNoOfLevels++;
+            // Calling countScenes() inside SceneHandler to find the new 
+            // filename to save our scene with. 
+            int currentNoOfLevels = SceneHandler.countScenes();
+            currentNoOfLevels++;
+            SetupCamera();
 
-			// Setting up the camera.
-			GameObject cameraGameObject = new GameObject("Main Camera");
-			cameraGameObject.AddComponent<Camera>();
-			cameraGameObject.tag = "MainCamera";
+            LevelBuilderWindow.SetUpScene();
+            SetupLighting();
 
-			LevelBuilderWindow.SetUpScene ();
+            GameObject level = CreateEmptyLevel();
+            CreateTerrainForLevel(level);
 
-			// Setting up the light.
-			GameObject directionalLight = new GameObject ("Directional Light");
-			Light lightSettings = directionalLight.AddComponent<Light> ();
-			lightSettings.type = LightType.Directional;
+            if (!EditorSceneManager.SaveScene(newScene, "Assets/Scenes/Levels/Level" + currentNoOfLevels + ".unity"))
+            {
+                Debug.Log("The savescene line failed!!!!!!!!!!!!!");
```

Comment:
:joy: can we add more details to help like isn't there a specific error that would tell what went wrong?

Reply:
Agreed (I didn't write this also) I think I'm just going to delete the check as we aren't using it right now

Reply:
I would not delete the check but throw an Exception instead?